### PR TITLE
Add conformance CI, update and fix tests

### PR DESCRIPTION
**Pull request by mvdbeek**

### CODE REVIEW

File: lib/galaxy/job_execution/output_collect.py [hunk 1]
```diff
@@ -593,15 +593,16 @@ def default_exit_code_file(files_dir, id_tag):
 
 def collect_extra_files(object_store, dataset, job_working_directory):
     file_name = dataset.dataset.extra_files_path_name_from(object_store)
-    temp_file_path = os.path.join(job_working_directory, "working", file_name)
+    output_dir = "working" if os.path.exists(os.path.join(job_working_directory, "working", file_name)) else "outputs"
```
Comment (jmchilton) [inline lib/galaxy/job_execution/output_collect.py:596]: I was going to augment tools with information about how to annotate this 😆 - a whole day project. This is so much easier, really awesome!

Comment (mvdbeek) [reply lib/galaxy/job_execution/output_collect.py:596]: We should ultimately switch everything, and then keep that logic in for a release or two :).

File: run_tests.sh [hunk 2]
```diff
@@ -675,10 +675,11 @@ if [ "$test_script" = 'pytest' ]; then
         coverage_arg="--cov-report term --cov=lib"
     fi
     if [ -n "$marker" ]; then
-        "$test_script" -v --html "$report_file" $coverage_arg  $xunit_args $extra_args -m "$marker" "$@"
+        marker_arg="-m '$marker'"
     else
-        "$test_script" -v --html "$report_file" $coverage_arg  $xunit_args $extra_args "$@"
+        marker_arg=""
     fi
+    "$test_script" -v --html "$report_file" $coverage_arg $xunit_args $extra_args $marker_arg "$@"
```
Comment (mvdbeek) [inline run_tests.sh]: I think I've spent an hour yesterday trying everything I could think of to get `"not cwl_conformance"` as an argument. If you don't quote this you get ` -m ''\''not' 'cwl_conformance'\'''`, if you do you get `'-m '\''not cwl_conformance'\'''`.

File: test/unit/tools/cwl_tools/conformance_to_test_cases.py [hunk 3]
```diff
@@ -163,6 +192,7 @@ class CwlConformanceTestCase(BaseCwlWorklfowTestCase):
         "valuefrom_ignored_null",
     ],
     "v1.1": [
+        "v1_1_wf_simple",
```
Comment (jmchilton) [inline test/unit/tools/cwl_tools/conformance_to_test_cases.py]: Wrong syntax here.

**Comment by jmchilton** [comment]
Amazing - thanks to you both for this awesome work!

**Pull request merged**